chore: updated search to use createAmplifyGeocoder (#3505)
aws-amplify/docs: src/fragments/lib/geo/js/search.mdx

First, make sure you've provisioned a search index resource and configured your app using the instructions in either [Getting started](/lib/geo/getting-started) or [Use existing Amazon Location Service resources](/lib/geo/existing-resources) and you have already setup [displaying a map](/lib/geo/maps) in your application.

> To add a location search UI component to your map, you need to use the [maplibre-gl-geocoder](https://github.com/maplibre/maplibre-gl-geocoder) library with Amplify Geo's `maplibre-gl-js-amplify` package. `maplibre-gl-js-amplify` makes it easy to integrate MapLibre with Amplify Geo by exporting `createAmplifyGeocoder()` a utility function that return an instance of `maplibre-gl-geocoder` with some pre-defined settings.

Reviewer: ```suggestion
To add a location search UI component to your map, you can use the [maplibre-gl-geocoder](https://github.com/maplibre/maplibre-gl-geocoder) library. `maplibre-gl-js-amplify` package makes it easy to integrate `maplibre-gl-geocoder` with Amplify Geo by exporting a utility function `createAmplifyGeocoder()` that returns an instance of `maplibre-gl-geocoder` with some pre-defined settings and supports all the [options](https://github.com/maplibre/maplibre-gl-geocoder/blob/main/API.md#parameters) for customizing the UI component
```
